llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -566,8 +553,10 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
                    Ty.getElementType() != SrcTy.getElementType();
           },
           0, 1)
-      .clampNumElements(0, v2s32, v4s32)
-      .clampMaxNumElements(1, s64, 2);
+      .clampNumElements(1, v4s16, v8s16)
+      .clampNumElements(1, v2s32, v4s32)
+      .clampMaxNumElements(1, s64, 2)
+      .moreElementsToNextPow2(1);
```

Comment:
`moreElementsVector` does not exist for G_FCMP:
https://github.com/llvm/llvm-project/blob/d0d072710468316edbd4130e50f1146c5a6aca89/llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp#L5317

Reply:
Yeah I had added it, but it was still falling back due to the unsupported shifts. Now that those are in it's looking better.